tests/integration/query_helpers/test_query_docs.py

```diff
@@ -0,0 +1,126 @@
+"""Integration tests using light weight FAISS index."""
+import os
+
+import pytest
+from langchain_community.embeddings import HuggingFaceEmbeddings
+from langchain_community.vectorstores.faiss import FAISS
+from langchain_core.documents.base import Document
+
+from ols.src.query_helpers.query_docs import QueryDocs, RetrieveDocsExceptionError
+
+
+@pytest.fixture
+def setup_faiss():
+    """Documents stored as  up FAISS index."""
+    list_of_documents = [
+        Document(page_content="foo", metadata={"page": 1, "source": "adhoc"}),
+        Document(page_content="bar", metadata={"page": 1, "source": "adhoc"}),
+        Document(page_content="foo", metadata={"page": 2, "source": "adhoc"}),
+        Document(page_content="barbar", metadata={"page": 2, "source": "adhoc"}),
+        Document(page_content="foo", metadata={"page": 3, "source": "adhoc"}),
+        Document(page_content="bar burr", metadata={"page": 3, "source": "adhoc"}),
+        Document(page_content="foo", metadata={"page": 4, "source": "adhoc"}),
+        Document(page_content="bar bruh", metadata={"page": 4, "source": "adhoc"}),
+    ]
+    embeddings = HuggingFaceEmbeddings(model_name="all-MiniLM-L6-v2")
+
+    file_path = "./index/faiss_index"
+
+    if os.path.exists(file_path):
```

Comment:
is it ok to have two possible ways how unit test will run? It will be different on CI, I guess

Reply:
prob worth to create it in all cases?

Reply:
Think of it as a light weight db. So creating it for every case will slow the test down. So we create one index store and set of document embeddings in the store and then test all out cases. Is it helpful?